OptimizelySDK/OptimizelyJson.cs

```diff
@@ -0,0 +1,137 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using OptimizelySDK.Logger;
+using System;
+using System.Collections.Generic;
+using Newtonsoft.Json.Linq;
+using OptimizelySDK.ErrorHandler;
+using System.Linq;
+using Newtonsoft.Json;
+
+namespace OptimizelySDK
+{
+    public class OptimizelyJson
+    {
+        private ILogger Logger;
+        private IErrorHandler ErrorHandler;
+
+        private string Payload { get; set; }
+        private Dictionary<string, object> Dict { get; set; }
+
+        public OptimizelyJson(string payload, IErrorHandler errorHandler, ILogger logger)
+        {
+            try
+            {
+                ErrorHandler = errorHandler;
+                Logger = logger;
+                Dict = (Dictionary<string, object>)ConvertIntoCollection(JObject.Parse(payload));
+                Payload = payload;
+            }
+            catch (Exception exception)
+            {
+                logger.Log(LogLevel.ERROR, "Provided string could not be converted to map.");
+                ErrorHandler.HandleError(new Exceptions.InvalidJsonException(exception.Message));
+            }
+        }
+        public OptimizelyJson(Dictionary<string, object> dict, IErrorHandler errorHandler, ILogger logger)
+        {
+            try
+            {
+                ErrorHandler = errorHandler;
+                Logger = logger;
+                Payload = JsonConvert.SerializeObject(dict);
+                Dict = dict;
+            }
+            catch (Exception exception)
+            {
+                logger.Log(LogLevel.ERROR, "Provided map could not be converted to string.");
+                ErrorHandler.HandleError(new Exceptions.InvalidJsonException(exception.Message));
+            }
+        }
+
+        override public string ToString()
+        {
+            return Payload;
+        }
+
+        public Dictionary<string, object> ToDictionary()
+        {
+            return Dict;
+        }
+
+        /// <summary>
+        /// If JSON Data is {"k1":true, "k2":{"k3":"v3"}}
+        ///
+        /// Set jsonPath to "k2" to access {"k3":"v3"} or set it to "k2.k3" to access "v3"
+        /// Set it to null or empty to access the entire JSON data but type must be Dictionary<string, object> as generic type.
+        /// </summary>
+        /// <param name="jsonPath">Key path for the value.</param>
+        /// <returns>Value if decoded successfully</returns>
+        public T GetValue<T>(string jsonPath)
```

Comment:
Add a description above the examples in the function summary
`Returns the Optimizely Json data casted to type T if successful. if jsonPath is provided, then it will attempt to cast the value from that path.`